libs/metamask/src/lib/guard/wallet.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import { MetamaskService } from '../+state/metamask.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EthereumWalletGuard implements CanActivate {
+  constructor(
+    private router: Router,
+    private service: MetamaskService,
+  ) { }
+
+  async canActivate(): Promise<boolean | UrlTree> {
```

Comment:
```typescript
return !this.service.account;
```

Reply:
If I do it like this, when pressing F5, users will be redirected even if metamask is linked already